state/process.go

```diff
@@ -225,6 +222,39 @@ outer:
 					col.txnResults = append(col.txnResults, txnResult)
 					break
 				}
+				if spork.Version >= 8 {
+					// check if the first tx in the system collection contains events indicating scheduled transactions were run
+					// We are already on the slow path where GetTransactionsByBlockID/GetTransactionResultsByBlockID has failed
+					systemTxEvents := col.txnResults[len(col.txnResults)-1].Events
```

Comment:
```suggestion
					systemTxEvents := col.txnResults[0].Events
```
We are assuming here (I believe) that only the first transaction has been added to the list. I think it is better to make that explicit in the code.